cloudigrade/api/tasks/maintenance.py

```diff
@@ -308,3 +311,106 @@ def _delete_cloud_account_related_objects(cloud_account):
     )
     delete_machine_image_ids = set(machine_image_ids) - set(active_machine_image_ids)
     MachineImage.objects.filter(id__in=delete_machine_image_ids).delete()
+
+
+@shared_task(name="api.tasks.delete_orphaned_cloud_accounts")
+@aws.rewrap_aws_errors
+def delete_orphaned_cloud_accounts():
+    """
+    Identify and delete orphaned CloudAccount and *CloudAccount objects.
+
+    This function is a reactionary "cleanup" task to deal with broken data that we do
+    not understand how is persisting. Somehow it is possible to have a CloudAccount
+    without its cloud provider-specific instance (e.g. AwsCloudAccount), and that broken
+    relationship may cause problems such as a failing to configure new sources. Until we
+    confidently understand and prevent that situation, this function exists to clean up
+    the mess left behind.
+    """
+    max_updated_at = get_now() - timedelta(
+        seconds=settings.DELETE_ORPHANED_ACCOUNTS_UPDATED_MORE_THAN_SECONDS_AGO
+    )
+    _delete_orphaned_cloud_accounts(max_updated_at)
+    _delete_orphaned_cloud_account_content_objects(max_updated_at)
+
+
+def _delete_orphaned_cloud_accounts(max_updated_at):
+    """
+    Identify and delete orphaned CloudAccount objects (having no related *CloudAccount).
+
+    Args:
+        max_updated_at (datetime.datetime): max updated_id for finding potentially
+        affected instances so that we reduce the risk of deleting instances that are
+        still being created at the time that this function runs.
+    """
+    found_orphans = []
+    deleted_orphans = []
+    for cloud_account in CloudAccount.objects.filter(updated_at__lt=max_updated_at):
+        if not cloud_account.content_object:
+            found_orphans.append(cloud_account)
+            logger.info(
+                _("Found orphan %(cloud_account)s"), {"cloud_account": cloud_account}
+            )
+            try:
+                if source := sources.get_source():
```

Comment:
don't we need to pass the account number ?

Reply:
Indeed we do! I also spotted this locally but completely forgot to push it. Fixed now. 😅